File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 3/30/2026, 2:14:21 PM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        s = 0
        mx = nums[0]
        for i in range(len(nums)):
            s+=nums[i]
            if(s>mx): mx = s
            if(s<0): s = 0
        return mx

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 3/30/2026, 2:14:05 PM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxsum = nums[0],currsum = nums[0];
        for(int i = 1 ; i < nums.size() ; i++){
            currsum = max(nums[i],currsum+nums[i]);
            maxsum = max(maxsum,currsum);
        }
        return maxsum;
    }
};